x-pack/filebeat/input/httpjson/config.go

```diff
@@ -84,6 +83,14 @@ func (c *config) Validate() error {
 			}
 		}
 	}
+	if c.OAuth2 != nil {
+		if c.APIKey != "" || c.AuthenticationScheme != "" {
+			return errors.Errorf("invalid configuration: both oauth2 and api_key or authentication_scheme cannot be set simultaneously")
+		}
+		if err := c.OAuth2.validate(); err != nil {
```

Comment:
I think you can move the `OAuth2` validation into its own `Validate() error` function and go-ucfg should automatically invoke it when unpacking the config.